### PR TITLE
Bug 1739229: Correctly update kibana proxy related secrets and certs

### PR DESCRIPTION
**Pull request by ewolinetz**
Addresses:
  * https://bugzilla.redhat.com/show_bug.cgi?id=1739229
  * https://bugzilla.redhat.com/show_bug.cgi?id=1739226
  * https://bugzilla.redhat.com/show_bug.cgi?id=1739228

### CODE REVIEW

File: roles/openshift_logging_kibana/tasks/generate_proxy_secret_files.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    get_attributes: false
+    get_mime: false
+  register: oauth_secret_file
+
+# gen session_secret if necessary
+- name: Generate session secret
+  copy:
+    content: "{{ 32 | lib_utils_oo_random_word }}"
+    dest: "{{ generated_certs_dir }}/session_secret"
+  when: not session_secret_file.stat.exists or session_secret_file.stat.size > 50
```
Comment (richm) [inline roles/openshift_logging_kibana/tasks/generate_proxy_secret_files.yaml]: why 50?

Comment (ewolinetz) [reply roles/openshift_logging_kibana/tasks/generate_proxy_secret_files.yaml]: this was just copy and pasted to a new file so we could call it separate from the rest of the kibana install logic. its what we had before https://github.com/openshift/openshift-ansible/pull/11817/files#diff-eb8a7a231094de4642600a9bf48ce60fL87

File: roles/openshift_logging_kibana/tasks/generate_route.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  when: kibana_ca is not defined
+  changed_when: false
+
+- name: Generating Kibana route template
+  template:
+    src: "route_reencrypt.j2"
+    dest: "{{ tempdir }}/templates/kibana-route.yaml"
+  vars:
+    obj_name: "{{ kibana_name }}"
+    route_host: "{{ openshift_logging_kibana_hostname }}"
```
Comment (richm) [inline roles/openshift_logging_kibana/tasks/generate_route.yaml]: should this be the kibana ops hostname for the ops route?

**Comment by richm** [comment]
@ewolinetz I'm hoping https://github.com/openshift/origin-aggregated-logging/pull/1725 will solve these test flakes . . .

File: roles/openshift_logging_kibana/tasks/generate_route.yaml (hunk 2 above)
Comment (ewolinetz) [reply roles/openshift_logging_kibana/tasks/generate_route.yaml]: addressed here: https://github.com/openshift/openshift-ansible/pull/11817/files#diff-0857978473a58a6c671b308aece000d4R115-R118

**Comment by ewolinetz** [comment]
@richm it looks like the logging test is still failing from flakes

**Comment by richm** [comment]
yep - logging tests still flaky - hoping that https://github.com/openshift/origin-aggregated-logging/pull/1725 fixes them

**Pull request merged**